### PR TITLE
Moving over WIP work from fork

### PR DESCRIPTION
**Pull request by greymind**
cc @visweswaranr

### CODE REVIEW

**Comment by greymind** [comment]
@visweswaranr What are some of the caveats on this WIP implementation?

File: src/WebApiToTypeScript/Endpoints/AngularEndpointsService.cs [hunk 1]
```diff
@@ -8,63 +8,104 @@ public class AngularEndpointsService : ServiceAware
     {
         public TypeScriptBlock CreateServiceBlock()
```
Comment (dsperac) [inline src/WebApiToTypeScript/Endpoints/AngularEndpointsService.cs]: @visweswaranr @greymind we should discuss how to do this better, these files are not following single-responsibility anymore. If we want to support more frameworks / libraries in the future (for example, react), writing these if statements everywhere is gonna become hell. Probably best to create an additional layer of abstraction and then requesting instances depending on what implementation you want (angularjs, Angular, React..)

Comment (greymind) [reply src/WebApiToTypeScript/Endpoints/AngularEndpointsService.cs]: Yeah, agreed. I have some time to work on this now, and I'm absorbing this and writing it anew. Check the other branch ng-support-new! Added 3 service types for now - AngularJS, Angular and AXIOS. Working on Angular now, AXIOS to follow (for React and such).

Comment (visweswaranr) [reply src/WebApiToTypeScript/Endpoints/AngularEndpointsService.cs]: @dsperac ya agreed. Did that as a quick WIP thingy for NGV products. Balki will clean it up.

**Comment by visweswaranr** [comment]
@greymind A few things:
1. Its good for all files - endpoints, endpointservice, interface, etc to accept a list of custom imports
2. Option for switching between Observable/Promise

**Pull request merged**